Assets/HoloToolkit/Input/Scripts/GazeManager.cs

```diff
@@ -2,135 +2,81 @@
-        /// </summary>
-        public GameObject FocusedObject { get; private set; }
-
-        [Tooltip("Checking enables SetFocusPointForFrame to set the stabilization plane.")]
-        public bool SetStabilizationPlane = true;
-        [Tooltip("Lerp speed when moving focus point closer.")]
-        public float LerpStabilizationPlanePowerCloser = 4.0f;
-        [Tooltip("Lerp speed when moving focus point farther away.")]
-        public float LerpStabilizationPlanePowerFarther = 7.0f;
-
         private Vector3 gazeOrigin;
         private Vector3 gazeDirection;
-        private float lastHitDistance = 15.0f;
+        private float lastHitDistance = 4.0f;
 
         private void Update()
         {
             gazeOrigin = Camera.main.transform.position;
             gazeDirection = Camera.main.transform.forward;
 
             UpdateRaycast();
-
-            UpdateStabilizationPlane();
         }
 
         /// <summary>
         /// Calculates the Raycast hit position and normal.
         /// </summary>
         private void UpdateRaycast()
         {
-            // Get the raycast hit information from Unity's physics system.
-            RaycastHit hitInfo;
-            Hit = Physics.Raycast(gazeOrigin,
-                           gazeDirection,
-                           out hitInfo,
-                           MaxGazeDistance,
-                           RaycastLayerMask);
-
-            GameObject oldFocusedObject = FocusedObject;
             // Update the HitInfo property so other classes can use this hit information.
-            HitInfo = hitInfo;
+            HitInfo = InputModule.RaycastResult;
+            IsGazingAtHologram = HitInfo.isValid;
 
-            if (Hit)
+            if (this.IsGazingAtHologram)
             {
                 // If the raycast hits a hologram, set the position and normal to match the intersection point.
-                Position = hitInfo.point;
-                Normal = hitInfo.normal;
-                lastHitDistance = hitInfo.distance;
-                FocusedObject = hitInfo.collider.gameObject;
-            }
-            else
-            {
-                // If the raycast does not hit a hologram, default the position to last hit distance in front of the user,
-                // and the normal to face the user.
-                Position = gazeOrigin + (gazeDirection * lastHitDistance);
-                Normal = -gazeDirection;
-                FocusedObject = null;
-            }
+                Position = HitInfo.worldPosition;
+                Normal = HitInfo.worldNormal;
+                lastHitDistance = HitInfo.distance;
 
-            // Check if the currently hit object has changed
-            if (oldFocusedObject != FocusedObject)
-            {
-                if (oldFocusedObject != null)
+                if (Position == Vector3.zero)
                 {
-                    oldFocusedObject.SendMessage("OnGazeLeave", SendMessageOptions.DontRequireReceiver);
```

Comment:
Since this is a partial class you can choose to reuse the variables being exposed in GazeManager.cs already.